### PR TITLE
Fix tag names for Azure Linux images

### PR DESCRIPTION
**Pull request by sbomer**
The tag names should match the directory structure, and should include the architecture for arch-specific images.

### CODE REVIEW

**Review by akoeplinger — APPROVED** [review]

**Review by mthalman — COMMENTED** [review]
I like these changes. It brings consistency and clarity. But I think we need to update the documentation to reflect this pattern because it is a different pattern than what is currently used for non-cross images.

I'm specifically referring to the pattern documented at https://github.com/dotnet/dotnet-buildtools-prereqs-docker?tab=readme-ov-file#how-to-identify-an-image. The difference with the cross images is that the architecture specified refers to the cross target, not the architecture of the image.

**Comment by mthalman** [comment]
> Thanks! The docs currently say this about the arch:
> 
> > The tag format used by the images from this repository is mcr.microsoft.com/dotnet-buildtools/prereqs:<os-name>-<os-version>-<variant>-<architecture>
> > ...
> > <architecture> - Architecture of the OS (amd64 shall be implied if not specified).
> 
> > The folder structure used in [src](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src) aligns with the tagging convention
> 
> > If a Dockerfile is shared across multiple architectures, then the  folder should be omitted.
> 
> Personally I like the policy of allowing `-amd64` to be omitted from the tag format where it's implied, and keeping the simpler tags like `azurelinux-3.0-net8.0-crossdeps-builder`.
> 
> I propose we:
> 
> * Allow omitting `amd64` from the tag names _and_ folder structure, and undo [Move all Dockerfiles to arch-specific dir #1047](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1047) for images where we take advantage of this option.
> * Update docs to recommend the current pattern (`-<target>`) that we use for the cross-compilation images.
> 
> cc @richlander

I don't like removing it from the directory name. Because there are some Dockerfiles which are truly multi-arch and so they have no architecture directory. Saying that an amd64Dockerfile can get by without being contained in an amd64 directory causes confusion and ambiguity compared to the multi-arch ones.

For tags, I initially was ambivalent. But thinking more about it, I kind of feel it needs to be explicit because of the ambiguity that arises from it. One could imagine have multi-arch tags for our multi-arch Dockerfiles. And that would most definitely not have an arch in the tag name.

So I think the documented pattern should be changed to say that all architectures should be explicit both in tags and directories.

File: README.md [hunk 1]
```diff
@@ -12,20 +12,24 @@ The images produced from the Dockerfiles in this repository are published to the
 
 ## How to identify an image
 
-The tag format used by the images from this repository is `mcr.microsoft.com/dotnet-buildtools/prereqs:<os-name>-<os-version>-<variant>-<architecture>`
+There are two tag formats used by the images from this repository. Most of the imagces follow the format `mcr.microsoft.com/dotnet-buildtools/prereqs:<os-name>-<os-version>-<variant>-<architecture>`.
```
Comment (mthalman) [inline README.md:15]:
```suggestion
There are two tag formats used by the images from this repository. Most of the images follow the format `mcr.microsoft.com/dotnet-buildtools/prereqs:<os-name>-<os-version>-<variant>-<architecture>`.
```

File: README.md [hunk 2]
```diff
@@ -12,20 +12,24 @@ The images produced from the Dockerfiles in this repository are published to the
...
 
 Examples:
 
-- mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.20
+- mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.20-amd64
 - mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-helix-arm64v8
 
+The cross-compilation images follow the format `mcr.microsoft.com/dotnet-buildtools/prereqs:<os-name>-<os-version>-<variant>-cross-<target>`. These are all implicitly amd64 images.
+
+- `<target>` - Specifies the target for cross-compilation, including the targeted architecture and libc variant (glibc if not specified).
```
Comment (mthalman) [inline README.md:31]: I think providing some examples like the first format would be helpful here.

File: README.md [hunk 3]
```diff
@@ -165,8 +169,9 @@ From this commit of the `image-info.dotnet-dotnet-buildtools-prereqs-docker-main
 
 ### Source Folder Structure
 
-The folder structure used in [src](./src) aligns with the tagging convention - `<os-name>-<os-version>-<variant>-<architecture>`.
-For example, the Dockerfile used to produce the `mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.20` image is stored in the [src/alpine/3.20/amd64](./src/alpine/3.20/amd64) folder.
+The folder structure used in [src](./src) aligns with the tagging convention - `<os-name>-<os-version>-<variant>-<architecture>` or `<os-name>-<os-version>-<variant>-cross-<target>`.
+For example, the Dockerfile used to produce the `mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.20-amd64` image is stored in the [src/alpine/3.20/amd64](./src/alpine/3.20/amd64) folder.
+The Dockerfile used to produce the `mcr.`microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net8.0-cross-arm64` image is stored in the [src/azurelinux/3.0/net8.0/cross/arm64](./src/azurelinux/3.0/net8.0/cross/arm64) folder.
```
Comment (mthalman) [inline README.md:174]:
```suggestion
The Dockerfile used to produce the `mcr.microsoft.com/dotnet-buildtools/prereqs:azurelinux-3.0-net8.0-cross-arm64` image is stored in the [src/azurelinux/3.0/net8.0/cross/arm64](./src/azurelinux/3.0/net8.0/cross/arm64) folder.
```

**Review by mthalman — APPROVED** [review]

**Pull request merged**